src/module3/routers/group.ts

```diff
@@ -1,62 +1,70 @@
-import { Router, Request, Response } from "express";
+import { Router, Request, Response, NextFunction } from "express";
 import { groupSchema, groupUpdateSchema } from "../validation";
-import { validateSchema } from "../utils";
+import { validateSchema, BadRequestError } from "../utils";
 import { Group } from "../data-access";
 
 const groupRouter = Router();
 const groupDto = new Group();
 
-groupRouter.get('/groups', async (req: Request, res: Response) => {
-    const groups = await groupDto.findAll();
-    return res.json(groups);
+groupRouter.get('/groups', async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const groups = await groupDto.findAll();
+        return res.json(groups);
+    } catch (error) {
+        next(new BadRequestError(400, 'groups not found'));
+    }
 });
 
-groupRouter.get('/group/:id', async (req: Request, res: Response) => {
+groupRouter.get('/group/:id', async (req: Request, res: Response, next: NextFunction) => {
     const { id } = req.params;
     try {
         const group = await groupDto.findByPk(id);
         return res.json(group);
     } catch (error) {
-        console.log(error);
-        return res.status(400).json({ message: 'groupnot found' });
+        next(new BadRequestError(400, 'group not found'));
```

Comment:
Earlier implementation of error handling was very much fine too. With the current one, you are not making use of the error message. From services, you might receive the error for DB not reachable, etc or suppressing the information.
We can pass the error object to logger for our information purpose and send the generic error message back to the user.
Applies to all the next(new BadRequestError)